### ISSUE TITLE
Hercules will not build with static libraries

### ISSUE THREAD

**Issue by swhobbit** [Fish-Git/hyperion#113]
Hercules hyperion `./configure` complains it can't build with shared libraries on Raspbian (spinhawk, as i recall, does haver this issue)
So i ran the command:

```
./configure --enable-optimization="-mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard" --disable-multi-cpu --enable-bzip --enable-bzip2 --disable-shared --enable-static
```
Then I get the error:

```
ERROR: external-gui requires loadable modules (shared libraries = --enable-shared) 
configure: error: Please correct the above error(s) and try again 
```
I have zero use for a GUI, so:

./configure --enable-optimization="-mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard" --disable-multi-cpu --enable-bzip --enable-bzip2 --disable-shared --enable-static **--disable-external-gui**

Which does nothing.

***Expected behavior:***
At lease on of the following:

1. An explicitly environment which does not support an optional components like external GUI disables it rather than aborting the configure.
2. An explicit option such as `--disable-external-gui` is supported.

Under **no** conditions should building with static libraries be prevented.

**Comment by swhobbit** [comment]
This works on spinhawk:

./configure --enable-optimization="-mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard" --disable-multi-cpu  --disable-shared --enable-static  --enable-cckd-bzip2  --enable-het_bzip2

**Comment by swhobbit** [comment]
This also fails  on 
`FreeBSD frederica 11.1-RELEASE-p11 FreeBSD 11.1-RELEASE-p11 #0: Thu Jun 21 03:46:08 UTC 2018     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GENERIC  amd64`

Note how it ignores the --enabled-shared flag.

```[frederica:0,~/src/hyperion,12:22:57,21] ./configure --enable-shared 
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... autoconf/install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... nawk
  ...
checking for off_t... yes
checking for _LARGEFILE_SOURCE value needed for large files... no
checking whether getopt wrapper kludge is necessary... no
ERROR: external-gui requires loadable modules (shared libraries = --enable-shared) 
configure: error: Please correct the above error(s) and try again 
[frederica:0,~/src/hyperion,12:23:26,21] 
```

which does not enable-shared by default.   I'd suggest installing a VM with FreeBSD for testing.

**Comment by swhobbit** [comment]
btw, this affects the build on Raspberry Pi, which makes it more important than #11. (RPi is my primary Hercules platform.)

**Comment by Fish-Git** [comment]
> I'd suggest installing a VM with FreeBSD for testing.

I'm sorry but that is _**not**_ going to happen.

There are simply too many variations (different distributions) of *Nix out there for me to bother trying to install each one of them inside a virtual machine just so I can verify SDL Hyperion builds okay on each one. This is _especially_ true given that, as I mentioned in another GitHub Issue, I am _not_ a "*Nix person". I'm a Windows person. I know very little about Linux et al.

For a non-Linux person to have to go through the bother of installing and configuring an operating system they're unfamiliar and inexperienced with is quite frankly asking a bit too much.

Instead, I rely on the *Nix Hercules developer crowd (which I would include you in) to report whatever problems they find so I can fix them one by one on a "best effort" basis (or even better: fix it themselves and then simply offer me the patch!).

I have CentOS 6.10 (and 7.0) installed inside a virtual machine and even that's almost more than I can handle. For me to try install yet another (and _another_, and _**another**_, etc) variation of Linux ("*Nix", whatever) is simply too much.

Sorry!&nbsp;`:(`

**Comment by swhobbit** [comment]
> > I'd suggest installing a VM with FreeBSD for testing.

> I'm sorry but that is not going to happen.

*Shrug.* Have it your way.  

Of course, anyone working with any IBM OS finding that UNIX/Linux is foreign is hilarious to me.

If your Linux weren't already installed, I'd suggest Ubuntu rather than CentOS based on popularity. All the *BSD versions lost the OSS OS war to Linux years ago, of course -- I only suggested it because it more or less gives covers Mac as well.

me?  I'm multi-platform.  :-)

-ahd-

**Comment by Fish-Git** [comment]
> Of course, anyone working with any IBM OS finding that UNIX/Linux is foreign is hilarious to me.

What makes you think I work with any modern IBM OS? I may have an IBM OS background (grew up on DOS/VSE and VM/SP 5) but that was 30+ years ago! Today I pretty much know only enough about z/OS and z/VM to be barely able to IPL them having switched to Windows development when I left the mainframe world and had trouble finding a job.

So yeah, UNIX/Linux very much _**is**_ foreign to me! Why should that be hilarious?

**Comment by swhobbit** [comment]
I am unsure why Hercules was split into five packages, but such complications provide a sufficient additional load that my seemingly simple quest to use the LCS support has already consumed too much of my time. I'm closing my remaining issues and dropping all attempts to use Hyperion.

**Comment by Fish-Git** [comment]
> I'm closing my remaining issues and dropping all attempts to use Hyperion.

That's your choice of course, Drew, but I'm going to keep this issue open and try to resolve it anyway.

And regardless of your feelings towards this particular Hyperion, I want you to know I very much _**do**_ appreciate all of your valuable feedback! So thank you very much for that.

Take care. I wish you well and harbor no ill feelings.

**Comment by swhobbit** [comment]
> And regardless of your feelings towards this particular Hyperion

I'm not mad, just a tad frustrated.  Drop me a note at $firstname @ $lastname . us for a fuller explantation that we don't need to share with entire Interwebs.

**Comment by Fish-Git** [comment]
> I'm not mad, just a tad frustrated.

Oh I know you're not mad, Drew! And I understand your frustration all too well too!

But in my own experience frustration can be just a stone's throw away from being mad, too!&nbsp;`;-)`

> Drop me a note at $firstname @ $lastname . us for a fuller explanation that we don't need to share with entire Interwebs.

No need. I get it. My Hyperion support for static linking is broken/missing and the whole concept of my External Packages design doesn't sit well with some people as it introduces an extra burden (an extra step) on some platforms, when all the person wants to do is build Hercules! They don't want to have to "jump through hoops" just to build it. I get that.

So no worries. We're cool.&nbsp;`:)`

**Comment by wably** [comment]
Drew,

I encountered this very same issue on a Raspberry Pi3 a few weeks ago (sorry but I just saw this issue today).  I could not find a ./configure solution way around it.

Ultimately though, I did try the 1Stop-CMake build process that Steve Orso introduced into Hyperion last year.  That resolved the issue completely.  And by the way, Steve did a nice job with the new CMake process. Nice messaging and it just works.   It was so simple:  git clone the Hyperion respository to your Pi, and then from the directory where all of the source is, issue 1Stop-CMake and relax.  You do have to install CMake  (from CMake.org) if you don't have it.  The apt-get version of CMake on the Pi is too old; use the website.  Installing CMake was also a cinch; I used the .sh distribution.

I also migrated from Spinhawk to Hyperion because of LCS problems in Spinhawk.  That was why I needed to build with Hyperion.

Regards,
Bob

**Comment by swhobbit** [comment]
No, I've given up on networking under Hercules.  It won't get the 4361 networked, and to be frank, networking is not even in my current mandate for the 4361.

The Hercules developers should set aside their decade of differences and merge down to one branch or more branches on one busy (and often built) repository, and generally all push in the same direction. But I have no influence there.  Too many people want to rule in Hell than serve in heaven, I guess.
